When using a DNS cache and resolver (like dnsmasq for example) we can have more than MAXNS entries
in a custom resolv file (these entries can be the nameservers received from DHCP and/or added through
connman's configuration files).

We also tweak the order so that first will appear the nameservers set-up in connman's config files,
followed by the nameservers received from DHCP in the order in which they have been received.

Index: connman-1.30/src/resolver.c
===================================================================
--- connman-1.30.orig/src/resolver.c
+++ connman-1.30/src/resolver.c
@@ -115,9 +115,9 @@ static int resolvfile_export(void)
 	if (count)
 		g_string_append_printf(content, "\n");
 
-	for (count = 0, list = g_list_last(resolvfile_list);
-						list && (count < MAXNS);
-						list = g_list_previous(list)) {
+	for (list = g_list_first(resolvfile_list);
+						list;
+						list = g_list_next(list)) {
 		struct resolvfile_entry *entry = list->data;
 
 		if (!entry->server)
@@ -125,7 +125,6 @@ static int resolvfile_export(void)
 
 		g_string_append_printf(content, "nameserver %s\n",
 								entry->server);
-		count++;
 	}
 
 	old_umask = umask(022);
